clang/lib/Lex/Preprocessor.cpp

```diff
@@ -954,6 +954,9 @@ void Preprocessor::Lex(Token &Result) {
     }
   }
 
+  if (CurLexer && ++CheckPointCounter == 1024)
```

Comment:
We might want to make this configurable? At the very least, we should have some comments explaining the 1024.